TЗ - 11.

shepilovaElena opened the pull request:
Рефакторинг - хранилища InMemoryFilmStorage.java и InMemoryUserStorage.java, и интерфесы FilmStorage.java и UserStorage.java Добавление UserService.java и методов addToFriends(), deleteFromFriends(), getJointFriends(), getUserFriendsList() Добавление FilmService.java и методов addLike(), deleteLike(), getBestFilm() Обработка ошибок через ErrorHandler.java
Доработка контроллеров

shepilovaElena committed:
TЗ - 11.
Рефакторинг - хранилища InMemoryFilmStorage.java и InMemoryUserStorage.java, и интерфесы FilmStorage.java и UserStorage.java
Добавление UserService.java и методов addToFriends(), deleteFromFriends(), getJointFriends(), getUserFriendsList()
Добавление FilmService.java и методов addLike(), deleteLike(), getBestFilm()
Обработка ошибок через ErrorHandler.java
Доработка контроллеров

shepilovaElena committed:
TЗ - 11.
аннотация @Builder

shepilovaElena committed:
TЗ - 11.
аннотация @Builder правки

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -2,62 +2,92 @@
-        counter = counter + 1;
-        return counter;
-    }
+    FilmService filmService;
```
нужен модификатор private

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -2,62 +2,92 @@
+    }
+
+    @PutMapping("/{id}/like/{userId}")
+    public void addLike(@PathVariable long id, @PathVariable long userId) {
```
оставь в этом методе только вызов filmService.addLike(), а остальное перенеси в сервис

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -2,62 +2,92 @@
+        if (!checkId(id)) {
+            throw new NotFoundException("Фильм с id" + id + " не найден.");
+        }
+        if (!filmService.getUserStorage().getAllUsers().contains(filmService.getUserStorage().getUserById(userId))) {
```
очень не продуктивно выполнена проверка существования пользователя. Лучше сделать обычный get() пользователя и проверить его на null. И оформить это во внутренний метод, чтобы не повторять одну и ту же логику несколько раз

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -2,62 +2,92 @@
+        if (!filmService.getUserStorage().getAllUsers().contains(filmService.getUserStorage().getUserById(userId))) {
+            throw new NotFoundException("Пользователь с id" + userId + " не найден.");
+        }
+        filmService.addLike(id, userId);
```
добавь лог вызова метода

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -2,62 +2,92 @@
+    }
+
+    @DeleteMapping("/{id}/like/{userId}")
+    public void deleteLike(@PathVariable long id, @PathVariable long userId) {
```
аналогично для filmService.deleteLike()

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -2,62 +2,92 @@
+        if (!checkId(id)) {
+            throw new NotFoundException("Фильм с id" + id + " не найден.");
+        }
+        if (!filmService.getUserStorage().getAllUsers().contains(filmService.getUserStorage().getUserById(userId))) {
```
переиспользовать метод, о котором писал выше, только все это в классе FilmService

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -2,62 +2,92 @@
+        if (!filmService.getUserStorage().getAllUsers().contains(filmService.getUserStorage().getUserById(userId))) {
+            throw new NotFoundException("Пользователь с id" + userId + " не найден.");
         }
+        filmService.deleteLike(id, userId);
```
добавь лог вызова метода

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -2,62 +2,92 @@
+
+    @GetMapping("/popular")
+    public List<Film> getBestFilms(@RequestParam(defaultValue = "10") String count) {
+        return filmService.getBestFilms(count);
```
добавь лог вызова метода

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -2,62 +2,92 @@
+        }
     }
 
+    private boolean checkId(long id) {
```
лучше сделать обычный get() и проверить на null, чем вытаскивать всё хранилище и перебирать его

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/controller/UserController.java:
```diff
@@ -2,64 +2,108 @@
 public class UserController {
-    private Map<Integer, User> users = new HashMap<>();
-    private int counter = 0;
+    UserService userService;
```
нужен модификатор private

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/controller/UserController.java:
```diff
@@ -2,64 +2,108 @@
+    }
+
+    @PutMapping("/{id}/friends/{friendId}")
+    public void addToFriends(@PathVariable long id, @PathVariable long friendId) {
```
оставь в этом методе только вызов userService.addToFriends(), а остальное перенеси в сервис

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/controller/UserController.java:
```diff
@@ -2,64 +2,108 @@
+    @PutMapping("/{id}/friends/{friendId}")
+    public void addToFriends(@PathVariable long id, @PathVariable long friendId) {
+        if (!checkId(id)) {
+            throw new NotFoundException("Пользователь с id" + id + " не найден.");
```
выброс исключения с комментарием нужно перенести в метод, чтобы не повторять один и тот же текст

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/controller/UserController.java:
```diff
@@ -2,64 +2,108 @@
+        if (!checkId(friendId)) {
+            throw new NotFoundException("Пользователь с id" + friendId + " не найден.");
+        }
+        userService.addToFriends(id, friendId);
```
добавь лог вызова метода

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/controller/UserController.java:
```diff
@@ -2,64 +2,108 @@
+    }
+
+    @DeleteMapping("/{id}/friends/{friendId}")
+    public void  deleteFromFriends(@PathVariable long id, @PathVariable long friendId) {
```
аналогично для userService.deleteFromFriends()

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/controller/UserController.java:
```diff
@@ -2,64 +2,108 @@
+        if (!checkId(friendId)) {
+            throw new NotFoundException("Пользователь с id" + friendId + " не найден.");
+        }
+        userService.deleteFromFriends(id, friendId);
```
добавь лог вызова метода

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/controller/UserController.java:
```diff
@@ -2,64 +2,108 @@
+    }
+
+    @GetMapping("/{id}/friends")
+    public List<User> getUserFriendsList(@PathVariable long id) {
```
аналогично для userService.getUserFriendsList()

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/controller/UserController.java:
```diff
@@ -2,64 +2,108 @@
+        if (!checkId(id)) {
+            throw new NotFoundException("Пользователь с id" + id + " не найден.");
+        }
+        return userService.getUserFriendsList(id);
```
добавь лог вызова метода

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/controller/UserController.java:
```diff
@@ -2,64 +2,108 @@
+    }
+
+    @GetMapping("/{id}/friends/common/{otherId}")
+    public List<User> getJointFriends(@PathVariable long id, @PathVariable long otherId) {
```
аналогично для userService.getJointFriends()

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/controller/UserController.java:
```diff
@@ -2,64 +2,108 @@
+        if (!checkId(otherId)) {
+            throw new NotFoundException("Пользователь с id" + otherId + " не найден.");
+        }
+        return userService.getJointFriends(id, otherId);
```
добавь лог вызова метода

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/controller/UserController.java:
```diff
@@ -2,64 +2,108 @@
+        }
+    }
+
+    private boolean checkId(long id) {
```
лучше сделать обычный get() и проверить на null, чем вытаскивать всё хранилище и перебирать его

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/service/FilmService.java:
```diff
@@ -0,0 +1,70 @@
+
+@Service
+public class FilmService {
+    FilmStorage inMemoryFilmStorage;
```
всем параметрам нужны модификатор private

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/service/FilmService.java:
```diff
@@ -0,0 +1,70 @@
+        this.inMemoryUserStorage = inMemoryUserStorage;
+    }
+
+    public UserStorage getUserStorage() {
```
этот метод нужно удалить

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/service/UserService.java:
```diff
@@ -0,0 +1,59 @@
+
+    }
+
+    public List<User> getJointFriends(long user1Id, long user2Id) {
```
сначала вынеси пользователей в переменные, чтобы каждый раз в цикле за ними в хранилище не ходить

shepilovaElena committed:
TЗ - 11.
добавление логов.
перенос проверок в сервисы.
исправление методов checkUserId() и checkFilmId()

shepilovaElena committed:
TЗ - 11.
checkstyle
